### ISSUE TITLE
Repeat calls of `fastLinkJS` cause `IllegalStateException` on Windows and JDK 17+

### ISSUE THREAD

**Issue by nartamonov** [scala-js/scala-js#4841]
## How to reproduce

* Clone https://github.com/nartamonov/scalajs-bug-repr
* Enter SBT console
* `fastLinkJS` (should execute without errors), do not exit SBT console
* Make any changes to `Main.scala`
* `fastLinkJS` again, it throws exception:
    ```
    sbt:scalajs-bug-repr> fastLinkJS
    [info] compiling 1 Scala source to C:\Temp\scalajs-bug-repr\target\scala-2.13\classes ...
    [info] Fast optimizing C:\Temp\scalajs-bug-repr\target\scala-2.13\scalajs-bug-repr-fastopt
    Exception in thread "Thread-12" java.lang.IllegalStateException: Promise already completed.
      | => sat scala.concurrent.Promise.complete(Promise.scala:53)
            at scala.concurrent.Promise.complete$(Promise.scala:52)
            at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
            at scala.concurrent.Promise.success(Promise.scala:86)
            at scala.concurrent.Promise.success$(Promise.scala:86)
            at scala.concurrent.impl.Promise$DefaultPromise.success(Promise.scala:187)
            at org.scalajs.linker.PathOutputDirectory$Handler$5$.completed(PathOutputDirectory.scala:224)
            at org.scalajs.linker.PathOutputDirectory$Handler$5$.completed(PathOutputDirectory.scala:218)
            at java.base/sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:129)
            at java.base/sun.nio.ch.Invoker.invokeDirect(Invoker.java:160)
            at java.base/sun.nio.ch.Invoker.invoke(Invoker.java:188)
            at java.base/sun.nio.ch.Invoker.invoke(Invoker.java:300)
            at java.base/sun.nio.ch.WindowsAsynchronousFileChannelImpl$ReadTask.failed(WindowsAsynchronousFileChannelImpl.java:501)
            at java.base/sun.nio.ch.PendingIoCache$1.run(PendingIoCache.java:155)
            at java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:113)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
            at java.base/java.lang.Thread.run(Thread.java:1623)
    ```
* If second call of `fastLinkJS` was completed without errors, repeat again (change source code and execute cmd). After 2-3 times it should throw error.

## Environment

The bug is reproduced under the following conditions:

* Windows 10 22H2 build 19045.2604
* Eclipse Temurin 17.0.6+10 or 20+36

With JDK 11 or Linux all works fine.

**Comment by sjrd** [comment]
Hello, and thanks for the report.

I tried reproducing this on Windows 11 22H2 build 22621.1413, using Temurin-17+35 (build 17+35), but could not trigger the issue. I repeated the change+fastLinkJS step 20 times in a row without any issue. Is there anything else that could help me reproduce?

**Comment by sjrd** [comment]
I have a tentative "shot in the dark" fix over at #4842. @nartamonov Perhaps you could try it locally to see if it fixes your issue? You can find instructions on how to locally publish Scala.js at
https://github.com/scala-js/scala-js/blob/main/DEVELOPING.md#publish-locally

Note that due to other recent changes, the steps you provide will not even run into the situation you describe. You will have to `reload`+`fastLinkJS` at every step, in addition to changing the file. (in a single sbt session, the code path reported in the stack trace won't be used more than once).

**Comment by nartamonov** [comment]
@sjrd Yes, it's a strange bug, it reproduces on my work PC, but don't reproduces under the same circumstances on my home PC. Home PC is much more performant that the former, may be that's the reason.

Anyway, I've upgraded to ScalaJS 1.13.1 and the bug doesnt reproduces anymore on work PC! Thank you very much for quick fix! 👍 👍
